packages/aws-cdk-lib/aws-batch/lib/ecs-container-definition.ts

```diff
@@ -1087,6 +1092,13 @@ export class EcsFargateContainerDefinition extends EcsContainerDefinitionBase im
         operatingSystemFamily: this.fargateOperatingSystemFamily?._operatingSystemFamily,
       },
     };
+
+    // readonlyRootFilesystem isn't applicable to Windows, see https://kubernetes.io/docs/concepts/windows/intro/
+    if (this.fargateOperatingSystemFamily?.isWindows()) {
```

Comment:
I think I could change this line to use `undefined` instead of `false` ?

`this.readonlyRootFilesystem = props.readonlyRootFilesystem ?? undefined;` 

That would change the default value but that shouldn't matter because the default was `false` ? Then I can remove quite a bit of this code.

Reply:
@msambol If we set `this.readonlyRootFilesystem` from `false` to `undefined`, I am unsure what the consequences of that might be. I assume `undefined` will omit the value vs `false` setting this directly.

Now I think about it, I like the default being undefined but only if the service defaults to `false` when the property is not provided. Let me poke someone from the team that might know more to see what you suggest here will have wider side affects we might want to avoid.

Reply:
@msambol So one thing I missed, we can't update the `this.readonlyRootFilesystem` because that may break customers you expected the default value to be `false`. So I think we have to do it in the way you have it. If we can simplify it with a ternary in the `containerDef` I think that is helpful but not strictly required.

Reply:
makes sense... I updated the PR.